src/containers/Secure.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2017-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import * as React from 'react';
+import useOktaAuth from '../context/useOktaAuth';
+import useAuthRequired, { AuthRequiredOptions } from '../hooks/useAuthRequired';
+import useComponents, { ComponentsOptions } from '../hooks/useComponents';
+
+export type SecureComponentProps = React.PropsWithChildren<AuthRequiredOptions & ComponentsOptions>;
+
+const Secure: React.FC<SecureComponentProps> = ({
+  children,
+  ...options
+}) => {
+  const oktaContext = useOktaAuth();
+  const { isAuthenticated, loginError } = useAuthRequired(oktaContext, options);
+  const { Loading, ErrorReporter } = useComponents(oktaContext, options);
```

Comment:
I think this component should accept `Loading` and `Error` Components (Like `LoginCallback` does)

And maybe an `onError` function, to give greater control of error handling

Reply:
This component actually supports props `errorComponent` and `loadingElement`, like `<LoginCallback>`.

Maybe it's a confusing name for `options` variable (which is actually a props object)?

These props are passed to util `useComponents()` which can return appropriate Error and Loading components